### PR TITLE
Added http_proxy to the helm chart

### LINKED ISSUE #409: Ability to set http proxy in helm chart
The EFS Controller cannot communicate with AWS API when deployed by the helm chart. This is because my organization does not currently deploy VPC endpoints for EFS and requires internet traffic to be routed through a proxy. 

The [EBS CSI Driver Helm Chart](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/templates/controller.yaml#L97) allows the http proxy for the controller to be set. I would like to be able to do the same thing with the EFS CSI Driver's Helm Chart.

### PR DESCRIPTION
**Pull request by clarkdoan**
**Is this a bug fix or adding new feature?**
This PR adds the ability of setting proxy variables. Mimics the EBS CSI Driver's [implementation](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/templates/controller.yaml#L97). 

**What is this PR about? / Why do we need it?**
Closes #409 

**What testing is done?** 
This was tested by upgrading a release and by creating a new release.

### CODE REVIEW

File: charts/aws-efs-csi-driver/values.yaml [hunk 1]
```diff
@@ -34,6 +34,22 @@ sidecars:
 
 imagePullSecrets: []
 
+proxy: {}
+#  http_proxy:
+#  no_proxy:
+
+resources:
```
Comment (wongma7) [inline charts/aws-efs-csi-driver/values.yaml:41]: think this showed up becasue you had to rebase ,sorry about that, could you remove (the resources got moved to controller.resources so this field isn't used anymore)

**Comment by danielwegener** [comment]
Unfortunately @clarkdoan  does not seem to be available. Would it be (process-wise) possible if I handed in his commit+fix in a different PR?

File: charts/aws-efs-csi-driver/values.yaml (hunk 1 above)
Comment (clarkdoan) [reply charts/aws-efs-csi-driver/values.yaml]: @wongma7 I just pushed an update.

Comment (danielwegener) [reply charts/aws-efs-csi-driver/values.yaml]: @clarkdoan Hate to ask for it, but would you rebase it once more please? :)

**Comment by danielwegener** [comment]
@wongma7 sounds good for me :)

**Comment by MageshSrinivasulu** [comment]
@danielwegener  Is the issue fixed and do we have the proxy added to the helm chart?

**Pull request closed without merging**